### PR TITLE
Add settings for CI services

### PR DESCRIPTION
**Pull request by k-takata**
Now the official Vim repository is on GitHub, we can easily use many CI services.
This PR adds setting files for the following CI services:
- Travis CI
- Coveralls
- AppVeyor

These setting files were written for [vim-ci](https://github.com/vim-jp/vim-ci) by [vim-jp](https://github.com/vim-jp) members.

Here are the sample results:
- https://travis-ci.org/k-takata/vim/builds/77359110
- https://coveralls.io/builds/3406634
- https://ci.appveyor.com/project/k-takata/vim/build/1

Note: In order to use these settings, someone needs to set up accounts for the services.

### CODE REVIEW

**Comment by crazymaster** [comment]
I think that `$TEST` is unnecessary.

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+version: "{build}"
+
+clone_folder: c:\projects\vim-ci
```
Comment (crazymaster) [inline appveyor.yml]: Is this vim-jp/vim-ci's directory structure?

Comment (k-takata) [inline appveyor.yml]: Oops! This line is not needed. We can use the default folder.
I will update later.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      - python-dev
+      - python3-dev
+      - liblua5.1-0-dev
+      - lua5.1
+
+before_install:
+  - pip install --user cpp-coveralls
+
+script:
+  - ./configure --with-features=$FEATURES $CONFOPT --enable-fail-if-missing && make
```
Comment (crazymaster) [inline .travis.yml]: `make -j2` is faster than `make`.

Comment (k-takata) [inline .travis.yml]: Maybe, `make -j$(getconf _NPROCESSORS_ONLN)` is better.
I will update later.

**Comment by k-takata** [comment]
Updated.

**Comment by chrisbra** [comment]
Bram, what is your opinion on integrating this patch? I think it would be good to have this included, right?

**Comment by mattn** [comment]
:+1:

**Comment by koron** [comment]
:+1:

**Comment by k-takata** [comment]
Thank you for merging this.

Next step is setting up CI accounts.
I think @chrisbra can also set up the team account and the project on [Travis-CI](https://travis-ci.org/) and [Coveralls](https://coveralls.io/).
I don't know how to set up a team account on [AppVeyor](http://www.appveyor.com/). Maybe it's better to set up the project on an owner's personal account.

After the setup, it might be better to add badges to README.md.
E.g.

``` diff
diff --git a/README.md b/README.md
--- a/README.md
+++ b/README.md
@@ -1,5 +1,8 @@
 `README.md` for version 7.4 of Vim: Vi IMproved.

+[![Build Status](https://travis-ci.org/vim/vim.svg?branch=master)](https://travis-ci.org/vim/vim)
+[![Coverage Status](https://coveralls.io/repos/vim/vim/badge.svg?branch=master&service=github)](https://coveralls.io/github/vim/vim?branch=master)
+[![Build status](https://ci.appveyor.com/api/projects/status/<project-key>/branch/master?svg=true)](https://ci.appveyor.com/project/<user-name>/vim/branch/master)

 ## What is VIM ##

```

(`<project-key>` and `<user-name>` should be changed appropriately.)

**Comment by k-takata** [comment]
Why the comments on this PR are not redirected to vim_dev?
I can only see my first comment. https://groups.google.com/d/topic/vim_dev/M1jf47u8OhY/discussion

**Comment by crazymaster** [comment]
> I don't know how to set up a team account on AppVeyor. Maybe it's better to set up the project on an owner's personal account.

I found how to create Appveyor account for Github organizations.
http://help.appveyor.com/discussions/questions/1154-appveyor-account-for-github-organizations

**Comment by chrisbra** [comment]
> Why the comments on this PR are not redirected to vim_dev?

There was once a problem on githubs site, that resulted in the vim-dev user not getting notifications. This should be fixed now, but notifications for such old issues and PR are not generated. 
That should be fixed with newly created issues and PRs

I'll look into creating those team accounts for travis-ci, coveralls and appveyor.

**Comment by brammool** [comment]
Ken Takata wrote:

> Thank you for merging this.
> 
> Next step is setting up CI accounts.
> I think @chrisbra can also set up the team account and the project on [Travis-CI](https://travis-ci.org/) and [Coveralls](https://coveralls.io/).
> I don't know how to set up a team account on [AppVeyor](http://www.appveyor.com/). Maybe it's better to set up the project on an owner's personal account.

Apparently Christian has done something, I received requests from three
apps to access private data in the Vim project.  What I don't understand
is why private access is requested.  The apps should only use the public
data.

## 

I recommend ordering large cargo containers of paper towels to make up
whatever budget underruns you have.  Paper products are always useful and they
have the advantage of being completely flushable if you need to make room in
the storage area later.
                (Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by chrisbra** [comment]
Yes I did. Not sure, what is meant by private. I requested access to those services Appveyor, Appveyor CI (this one wants to access private repositories, which I think do not exist and access has not been requested, I don't even understand, why there are 2 Appveyor services), Coveralls and Travis-CI both of which do not request private data. 

In any case, access rights can be removed later by checking the services property in the settings section of the repository.

**Comment by chrisbra** [comment]
Screenshot of what has been requested:
appveyor
appveyor-ci
coveralls
travis-ci

**Comment by k-takata** [comment]
In my case, the following permissions are required:
- AppVeyor
  - Read org and team membership
  - Access user email addresses (read-only)
- AppVeyor CI
  - Full control of repository hooks
  - Read org and team membership
  - Access commit status
- Coveralls
  - Read org and team membership
  - Access commit status
  - Access user email addresses (read-only)
- Travis CI
  - Read org and team membership
  - Access commit status
  - Access deployment status
  - Access user email addresses (read-only)
  - Write repository hooks

AppVeyor CI doesn't need the "Access private repositories" permission. Some setting might be wrong.

**Comment by brammool** [comment]
Christian -

> Screenshot of what has been requested:
> appveyor
> appveyor-ci
> coveralls
> travis-ci

Most of that appears to be harmless.  Only Travis-ci requests "Write
repository hooks".  Don't know what that is but "write" seems
potentially harmful.
- Bram

## 

A radioactive cat has eighteen half-lives.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by koron** [comment]
@brammool

> Only Travis-ci requests "Write repository hooks".  Don't know what that is
> but "write" seems potentially harmful.

According to http://docs.travis-ci.com/user/github-oauth-scopes/

> Building a new repository on Travis CI is as easy as enabling it in your
> profile and pushing a new commit.
> 
> Updating the webhook required for us to be notified from GitHub on new
> commits or pull requests requires this API scope. Additionally, your user
> needs to have admin access to the repository you want to enable.

This page contains descriptions for other permissions.

**Comment by 2072** [comment]
@brammool

From what I understand `Write repository hooks` just means that you authorize the applications to add hooks to your repository. These can be actions that are triggered whenever the repository is modified by some action (commits, history rewrite, etc...).

More about git hooks: https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks

**Comment by chrisbra** [comment]
> Most of that appears to be harmless. 

So you do approve those requests?

**Comment by chrisbra** [comment]
Bram, please add those badges to the README.md:

```
[![Build Status](https://travis-ci.org/vim/vim.svg?branch=master)](https://travis-ci.org/vim/vim)
[![Coverage Status](https://coveralls.io/repos/vim/vim/badge.svg?branch=master&service=github)](https://coveralls.io/github/vim/vim?branch=master)
```

(haven't been able to get appveyor to see the vim repository yet)

**Comment by brammool** [comment]
Christian Brabandt wrote:

> Bram, please add those badges to the README.md:
> [Build Status](https://travis-ci.org/vim/vim)
> [Coverage Status](https://coveralls.io/github/vim/vim?branch=master)

Done.  Noticed that I forgot to include the test file change in 7.4.879.

> (haven't been able to get appveyor to see the vim repository yet)

## 

hundred-and-one symptoms of being an internet addict:
10. And even your night dreams are in HTML.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by crazymaster** [comment]
Hi,

Travis-ci builds even if you push tags.
So, please add this code to the .travis.yml not to build if you push tags.

```
branches:
  except:
    - /^v[0-9]/
```

Regards,
Kenichi Ito

**Comment by brammool** [comment]
Kenichi Ito wrote:

> Travis-ci builds even if you push tags.
> So, please add this code to the .travis.yml not to build if you push tags.
> 
> ```
> branches:
>   except:
>     - /^v[0-9]/
> ```

I'll include it, thanks.

## 

hundred-and-one symptoms of being an internet addict:
13. You refer to going to the bathroom as downloading.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by k-takata** [comment]
@brammool @chrisbra 
Is there any progress with appveyor setting?

**Comment by chrisbra** [comment]
@k-takata appveyor has not been granted access to the vim repository. Have requested it again. @brammool  please approve.

**Comment by chrisbra** [comment]
Okay, I think I have configured appveyor. I am not sure, if this now runs automatically, whenever a commit is made, but we'll see soon. @brammool  Please include this badge to README.md

```
[![Appveyor Build status](https://ci.appveyor.com/api/projects/status/o2qht2kjm02sgghk?svg=true)](https://ci.appveyor.com/project/chrisbra/vim)
```

**Comment by crazymaster** [comment]
> https://ci.appveyor.com/project/chrisbra/vim

Why not under vim/vim?

**Comment by chrisbra** [comment]
> Why not under vim/vim?

That's because only the project owner can do that. I don't think this is a problem however, it tests the vim/vim master branch.

**Comment by k-takata** [comment]
Thank you all.
Now we can see the results of Travis-CI, Coveralls and AppVeyor on the top page.
vim-readme

I'm so glad about this.

**Pull request closed without merging**